### PR TITLE
Fix/secondary priors tweak

### PR DESCRIPTION
**Pull request by t-brandt**
In the last iteration I undid the 1/M secondary mass prior if the priors object was set.  Now I undo it only if a given secondary mass prior is set with finite variance.

### CODE REVIEW

File: orvara/main.py [hunk 1]
```diff
@@ -204,7 +204,9 @@ def lnprob(theta, returninfo=False, RVoffsets=False, use_epoch_astrometry=False,
         orbit.calc_offsets(data, params, model, i)
 
         if priors is not None:
-            lnp = lnp - 0.5*(params.msec - priors.get(f'm_secondary{i}', 1))**2/priors.get(f'm_secondary{i}_sig', np.inf)**2 + np.log(params.msec)
+            chisq_sec = (params.msec - priors.get(f'm_secondary{i}', 1))**2/priors.get(f'm_secondary{i}_sig', np.inf)**2
+            if chisq_sec > 0:
+                lnp = lnp - 0.5*chisq_sec + np.log(params.msec)
```
Comment (gmbrandt) [inline orvara/main.py:209]: @t-brandt is the desired behavior here that if m_secondary{i} is not set, then:
lnp = lnp -1/2 * Msec**2 - ln(1/Msec)   ?
What is the role of -1/2 * Msec**2  ? Should it just be lnp = lnp - ln(1/Msec)    if no msec prior is set?

Comment (t-brandt) [reply orvara/main.py:209]: This is already in the last line orbit.pyx in the priors:
`return log(sin(par.inc)*1./(par.sau*par.msec))`
so if no msec prior is set, we want to just do nothing (I think), which preserves the behavior above.  If an msec prior is set, then 
`lnp = lnp + np.log(params.msec)`
should undo the operation in orbit.pyx.

Comment (gmbrandt) [reply orvara/main.py:209]: ohh I get this now. I suggest making the intent clearer, perhaps using:
```
        if priors is not None:
            prior_exists_on_this_secondary = f'm_secondary{i}' in priors
            if prior_exists_on_this_secondary:
                chisq_sec = (params.msec - priors[f'm_secondary{i}'])**2/priors[f'm_secondary{i}_sig']**2
                lnp = lnp - 0.5*chisq_sec + np.log(params.msec)
```

That also kills the .get(key, fallback) logic.

Comment (gmbrandt) [reply orvara/main.py:209]: which will also be concidentally less logic calls (just one membership check instead of two checks + a value comparison)

**Pull request merged**